main.star

```diff
@@ -123,47 +156,217 @@ def instantiate_waku_nodes(waku_topology, same_toml_configuration):
+
+    artifact_id = render_templates(
+        config={
+            "targets.json": struct(
+                template=template,
+                data=template_data,
+            )
+        }
+    )
+
+    return artifact_id
+
+
+def set_up_prometheus(services):
+    # Create targets.json
+    targets_artifact_id = create_prometheus_targets(services)
+
+    # Set up prometheus
+    CONFIG_LOCATION = "/test"
+    CONFIG_LOCATION2 = "/tmp"
+    artifact_id = upload_files(
+        src=PROMETHEUS_CONFIGURATION_PATH
+    )
+
+    prometheus_service = add_service(
+        service_id="prometheus",
+        config=struct(
+            image=PROMETHEUS_IMAGE,
+            ports={
+                PROMETHEUS_PORT_ID: PortSpec(number=9090, transport_protocol="TCP")
+            },
+            files={
+                CONFIG_LOCATION: artifact_id,
+                CONFIG_LOCATION2: targets_artifact_id
+            },
+            cmd=[
+                "--config.file=" + CONFIG_LOCATION + "/prometheus.yml"
             ]
+        )
+    )
+
+    return prometheus_service
+
+
+def set_up_graphana(prometheus_service):
+    # Set up grafana
+    CONFIGURATION_GRAFANA = "/etc/grafana/"
+    DASHBOARDS_GRAFANA = "/var/lib/grafana/dashboards/"
+    CUSTOMIZATION_GRAFANA = "/usr/share/grafana/"
+
+    config_id = upload_files(
+        src=GRAFANA_CONFIGURATION_PATH
+    )
+    customization_id = upload_files(
+        src=GRAFANA_CUSTOMIZATION_PATH
+    )
+    dashboard_id = upload_files(
+        src=GRAFANA_DASHBOARD_PATH
+    )
+
+    prometheus_url = prometheus_service.ip_address + ":" + str(prometheus_service.ports[PROMETHEUS_PORT_ID].number)
+    prometheus_info = {"prometheus_url": prometheus_url}
+
+    # template
+    template = """
+        apiVersion: 1
+        datasources:
+            - name: Prometheus
+              type: prometheus
+              access: proxy
+              org_id: 1
+              url: http://{{.prometheus_url}}
+              is_default: true
+              version: 1
+              editable: true
+    """
+
+    artifact_id = render_templates(
+        config={
+            "datasources.yaml": struct(
+                template=template,
+                data=prometheus_info,
+            )
         }
-    }
+    )
+
+    grafana_service = add_service(
+        service_id="grafana",
+        config=struct(
+            image=GRAFANA_IMAGE,
+            ports={
+                GRAFANA_PORT_ID: PortSpec(number=GRAFANA_TCP_PORT, transport_protocol="TCP")
+            },
+            files={
+                CONFIGURATION_GRAFANA: config_id,
+                # customization_id: CUSTOMIZATION_GRAFANA,
+                DASHBOARDS_GRAFANA: dashboard_id,
+                "/etc/grafana/provisioning/datasources/": artifact_id
+            }
+        )
+    )
+
+    return grafana_service
 
-    services = instantiate_waku_nodes(decoded, same_toml_configuration)
 
-    interconnect_waku_nodes(decoded, services)
+def run(args):
```

Comment:
We should probably stop and delete the enclave before or it will break if it already exists

Reply:
Yes, this is something I asked to the Kurtosis team, because I don't think there is a way to do it with Starlark. So either you make sure to do it, or you don't give the enclave an ID right now.